pkg/storage/tsdb/block/block.go

```diff
@@ -338,9 +338,8 @@ func MarkForNoCompact(ctx context.Context, logger log.Logger, bkt objstore.Bucke
 	}
 
```

Comment:
this function now logs `requested to mark for no compaction, but file already exists; this should not happen; investigate` when you try to mark a block twice. I think this is confusing. 

One way to rework is to return a special sentinel error from the block package and handle that differently in the store-gateway and compactor packages.

Reply:
I might not making it an error, this is a warning log, the remark is not crucial failure - the current compactor should continue the compaction, not trigger any alert because of this. Since it would just end up on a no-op not worthy an error. The best I can do is to pass a errorhandler function, for compactor log warning and return nil error, but for store-gateway return error on frontend. But does it worth it for the code change?

Reply:
can we just return an error which the store-gateway ignores and the compactor always logs. Something like


```go 
var MarkExists = fmt.Errorf("mark exists")

func MarkForNoCompact() error {
	...
	if noCompactMarkExists {
		return MarkExists
	}
	...
}
```

compactor:
```go
if errors.Is(err, block.MarkExists) {
	log("requested to mark for no compaction, but file already exists; this should not happen; investigate")
}
```

and 

store-gateway:
```go
if !errors.Is(err, block.MarkExists) {
	log("couldn't mark block for no compaction")
}
```

Reply:
I personally don't like to return error and abandon the mark for following blocks just because of a no-op, but I don't want to block the PR just for this. updated as requested.

Reply:
> return error and abandon the mark for following blocks just because of a no-op

that's a good point, I don't think we should be changing the behaviour. How about we use a multiError to collect these errors? [for example here](https://github.com/grafana/mimir/blob/15ddfa77995aec1285ccb5f217861dbc094ebfe4/pkg/ruler/ruler.go#L1286)

Reply:
I like this multiError, added the change and + a small improvement on frontend.